integration_test.go

```diff
@@ -3848,3 +3848,64 @@ func TestTentacleCertificateResource(t *testing.T) {
 		return nil
 	})
 }
+
+func TestKubernetesAgentWorkerResource(t *testing.T) {
```

Comment:
I think you'd want a test for the agent as worker data_source too. Have a look at the other integration tests that have a part two to them (e.g. 2-projectgroup and 2a-projectgroupds). The part two is for testing the data_source.